src/pk/ecc/ecc_dp_find_by_name.c

```diff
@@ -0,0 +1,30 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+#include "tomcrypt.h"
+
+#ifdef LTC_MECC
+
+ltc_ecc_set_type* ecc_dp_find_by_name(char *curve_name)
+{
+   int i;
+
+   for (i = 0; ltc_ecc_sets[i].size != 0; i++) {
+      if (ltc_ecc_sets[i].name != NULL && XSTRCMP(ltc_ecc_sets[i].name, curve_name) == 0) {
+         return (ltc_ecc_set_type*)&ltc_ecc_sets[i];
```

Comment:
what I meant was "why is _this_ `return `__`(ltc_ecc_set_type*)`__`&ltc_ecc_sets[i];` cast required?"

and I just got the answer... because the API signature is lying and you're returning a const object as non-const

```
src/pk/ecc/ecc_dp_find_by_name.c: In function ‘ecc_dp_find_by_name’:
src/pk/ecc/ecc_dp_find_by_name.c:20:17: warning: return discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
          return &ltc_ecc_sets[i];
```
three possibilities to solve this in order of preference:
1. make the return value a `const ltc_ecc_set_type*`
1. make a copy of the object you want to return before returning it as non-const (which the caller then has to free)
1. make `ltc_ecc_sets` not const